modules/navigation/godot_navigation_server.cpp

```diff
@@ -680,14 +680,22 @@ COMMAND_4(agent_set_callback, RID, p_agent, Object *, p_receiver, StringName, p_
 	RvoAgent *agent = agent_owner.get_or_null(p_agent);
 	ERR_FAIL_COND(agent == nullptr);
 
-	agent->set_callback(p_receiver == nullptr ? ObjectID() : p_receiver->get_instance_id(), p_method, p_udata);
+	if (agent == nullptr) {
+		return;
+	}
 
-	if (agent->get_map()) {
-		if (p_receiver == nullptr) {
+	if (p_receiver == nullptr) {
+		agent->set_callback(ObjectID(), p_method, p_udata);
+		if (agent->get_map()) {
 			agent->get_map()->remove_agent_as_controlled(agent);
-		} else {
-			agent->get_map()->set_agent_as_controlled(agent);
 		}
+		return;
+	}
+
+	agent->set_callback(p_receiver->get_instance_id(), p_method, p_udata);
```

Comment:
`p_receiver` won't match the null check since it's a wild pointer. So the problem still exists.

Probably it should be passed as an `ObjectID` instead of `Object *`, or just use `Callable` as the parameter.

Reply:
ObjectID sounds fine.